File: PHRPReader/clsScanStatsInfo.vb
Option Strict On

Public Class clsScanStatsInfo

    Private ReadOnly mScanNumber As Integer
    Private mScanTypeName As String

    ' ReSharper disable once ConvertToVbAutoProperty
    Public ReadOnly Property ScanNumber As Integer
        Get
            Return mScanNumber
        End Get
    End Property

    Public Property ScanTimeMinutes As Single

    Public Property ScanType As Integer

    Public Property TotalIonIntensity As Double

    Public Property BasePeakIntensity As Double

    Public Property BasePeakMZ As Double

    Public Property BasePeakSignalToNoiseRatio As Double

    Public Property IonCount As Integer

    Public Property IonCountRaw As Integer

    Public Property ScanTypeName As String
        Get
            Return mScanTypeName
        End Get
        Set(value As String)
            If String.IsNullOrEmpty(value) Then
                mScanTypeName = String.Empty
            Else
                mScanTypeName = value
            End If
        End Set
    End Property

    Public Sub New(intScanNumber As Integer, sngScanTimeMinutes As Single, intScanType As Integer)
        mScanNumber = intScanNumber
        ScanTimeMinutes = sngScanTimeMinutes
        ScanType = intScanType
    End Sub
End Class

File: PHRPReader/clsScanStatsInfo.cs
namespace PHRPReader
{
    public class clsScanStatsInfo
    {
        private readonly int mScanNumber;

        private string mScanTypeName;
        // ReSharper disable once ConvertToAutoProperty
        public int ScanNumber => mScanNumber;

        public float ScanTimeMinutes { get; set; }

        public int ScanType { get; set; }

        public double TotalIonIntensity { get; set; }

        public double BasePeakIntensity { get; set; }

        public double BasePeakMZ { get; set; }

        public double BasePeakSignalToNoiseRatio { get; set; }

        public int IonCount { get; set; }

        public int IonCountRaw { get; set; }

        public string ScanTypeName
        {
            get => mScanTypeName;
            set => mScanTypeName = string.IsNullOrEmpty(value) ? string.Empty : value;
        }

        public clsScanStatsInfo(int intScanNumber, float sngScanTimeMinutes, int intScanType)
        {
            mScanNumber = intScanNumber;
            ScanTimeMinutes = sngScanTimeMinutes;
            ScanType = intScanType;
        }
    }
}
